packages/cli/src/commands/repo/list-deprecations.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import chalk from 'chalk';
+import { ESLint } from 'eslint';
+import { Command } from 'commander';
+import { join as joinPath, relative as relativePath } from 'path';
+import { paths } from '../../lib/paths';
+import { PackageGraph } from '../../lib/monorepo';
+
+export async function command(cmd: Command) {
+  const packages = await PackageGraph.listTargetPackages();
+
+  const eslint = new ESLint({
+    cwd: paths.targetDir,
+    overrideConfig: {
+      plugins: ['deprecation'],
+      rules: {
+        'deprecation/deprecation': 'error',
+      },
+      parserOptions: {
+        project: [paths.resolveTargetRoot('tsconfig.json')],
+      },
+    },
+    extensions: ['jsx', 'ts', 'tsx', 'mjs', 'cjs'],
+  });
+
+  const { stderr } = process;
+  if (stderr.isTTY) {
+    stderr.write('Initializing TypeScript...');
+  }
+
+  const deprecations = [];
+  for (const [index, pkg] of packages.entries()) {
+    const results = await eslint.lintFiles(joinPath(pkg.dir, 'src'));
+    for (const result of results) {
+      for (const message of result.messages) {
+        if (message.ruleId !== 'deprecation/deprecation') {
+          continue;
+        }
+
+        const path = relativePath(paths.targetRoot, result.filePath);
+        deprecations.push({
+          path,
+          message: message.message,
+          line: message.line,
+          column: message.column,
+        });
+      }
+    }
+
+    if (stderr.isTTY) {
```

Comment:
Hehe shouldn't you put this before the actual work

Reply:
It's to have the `Initializing TypeScript...` text stick around until we've scanned the first package, since that takes a big chunk of time and there's no real way to find out when it's done